File: console/src/types/organization.ts
/**
 * Represent an organization.
 */
export type Organization = {
  /**
   * The organization id.
   */
  id: string;

  /**
   * The organization name.
   */
  name: string;
}

/**
 * The organization form creation/update value.
 */
export type OrganizationFormValue = Pick<
  Organization,
  'name'
>;

File: controlplane/frn-core/src/models/organization.rs
use database::{Factory, Persistable, Repository};
use sqlx::prelude::FromRow;
use sqlx::types::chrono;
use uuid::Uuid;

#[derive(Debug, Default, Factory, FromRow, Repository)]
pub struct Organization {
    /// The organization id
    #[repository(primary)]
    pub id: Uuid,
    /// The organization name
    pub name: String,
    /// Creation time of the organization
    pub created_at: chrono::DateTime<chrono::Utc>,
    /// Last update time of the organization
    pub updated_at: chrono::DateTime<chrono::Utc>,
}
